src/analyz/mod.rs

```diff
@@ -331,7 +318,8 @@ impl<'tcx> ToJson<'tcx> for mir::Operand<'tcx> {
 impl<'tcx> ToJson<'tcx> for mir::Constant<'tcx> {
     fn to_json(&self, mir: &mut MirState<'_, 'tcx>) -> serde_json::Value {
         json!({
-            "ty": self.ty.to_json(mir),
+            // FIXME remove
+            "ty": self.literal.ty.to_json(mir),
```

Comment:
Are types no longer parts of constants? Or are they part of the literal field?

Reply:
Yes, the `literal` field contains the `ty` now.

I looked at the mir-verifier parser for this case, and it looks like it could use some significant cleanup.  It handles three different `Literal` variants, but the current version of mir-json only ever emits one.  I'll make a separate branch for JSON cleanup after this, to cover `Literal` and the other JSON format FIXMEs you noted.